sdk/core/azure-core/azure/core/pipeline/policies/redirect.py

```diff
@@ -141,7 +141,7 @@ def increment(self, settings, response, redirect_location):
             response.http_request.method = 'GET'
         for non_redirect_header in self._remove_headers_on_redirect:
             response.http_request.headers.pop(non_redirect_header, None)
-        return settings['redirects'] > 0
+        return settings['redirects'] >= 0
```

Comment:
FWIW when I first saw this I expected it to count up, not count down. Maybe `redirects_left` or `redirects_remaining` if there is ever an opportunity to change it.

Reply:
I see what you mean. The redirect/retry policies both have a method called increment which actually decrements the counters.